### DISCUSSION TITLE
How to create a custom iso

### DISCUSSION THREAD

**Discussion by burggraf** [Darin755/browser-linux#7] [General]
Super cool repo -- awesome work here!

How can I go about creating a custom ISO with specific packages installed?  I want to get PostgreSQL installed by default on a minimum install.

**Comment by burggraf** [comment]
@Darin755 I've tried overlaying the minimal files with https://github.com/Darin755/browser-linux and I got an image to boot, but it's not building a full v86-linux.iso file?

**Reply by jmezzacappa** [reply]
It may be worth checking your configuration with Buildroot's `make menuconfig`. The iso image settings would be under the "Filesystem images" section.

**Reply by burggraf** [reply]
What settings should I be looking for here?  I was able to get postgres into the image, that looks like it's working, so that's cool.

**Reply by jmezzacappa** [reply]
I don't remember the exact setting, but I'm pretty sure there are not very many iso-related settings in there, so it should be easy to find, IIRC.

**Reply by burggraf** [reply]
It looks like "iso image" was already selected, with bootloader set to isolinux, and it is booting, it just stops during the boot process.  Let me post an image of what the boot looks like:
<img width="881" alt="Screen Shot 2022-08-29 at 5 54 12 AM" src="https://user-images.githubusercontent.com/225717/187224860-379b719e-9bcb-4376-abf1-0924509cd8d8.png">

**Reply by jmezzacappa** [reply]
Ah, I see. I think I have encountered similar errors at some point. There were quite a few along the way, so I don't remember the specific errors/solutions 😅 Several of them were related to the kernel configuration, IIRC.

(Edit to add some clarification: I was not using this repo, so I don't want to imply that there are issues with any kernel configs here 🙂)

**Comment by jmezzacappa** [comment]
If the goal is only to run Postgres in the browser, I wrote a guide for a simpler approach here: https://www.crunchydata.com/blog/crazy-idea-to-postgres-in-the-web-browser

(I hope it is not impolite to post that link here. I can delete this comment, if desired.)

**Reply by burggraf** [reply]
Ah, that was your article, cool!  Awesome work.  I'm trying to do this with buildroot and a very minimal config.

**Reply by burggraf** [reply]
BTW: I could not get your guide to work, it fails with: boot failed, not a bootable disk.  Maybe because I'm on an M1/arm processor?

**Reply by burggraf** [reply]
Here's what I get on your guide:
<img width="832" alt="Screen Shot 2022-08-29 at 7 34 04 AM" src="https://user-images.githubusercontent.com/225717/187226226-aa4a6ade-f6cd-44e3-8b6b-5f4d7ddea712.png">

**Reply by jmezzacappa** [reply]
Oh no! What command did you use to run qemu?

**Reply by burggraf** [reply]
I just copied your guide exactly.  I even tried an updated version of `alpine-virt-3.16.0-x86.iso`, but it had the same issue.
The qemm command was:

```
qemu-system-x86_64 \
  -m 256M \
  -cdrom alpine-virt-3.16.0-x86.iso \
  -drive file=alpine.img,format=raw
  
```

**Reply by jmezzacappa** [reply]
Thanks for letting me know. I was using an Intel Mac when I wrote the guide. I will try to reproduce the error on my M1 when I get time.

**Reply by burggraf** [reply]
Let me know if there's anything you want me to try.  Happy to reproduce it (I just did when this thread came up and it's the same issue).

**Comment by burggraf** [comment]
@Darin755 seems to have all the files necessary for the configuration, I just don't have everything in the right place, I don't think.

**Reply by burggraf** [reply]
@jmezzacappa our intention is to create an open-source version of this so if that doesn't fit into your business model and you want to opt out of helping here, I understand.  I don't want to step on any toes. :)

**Reply by jmezzacappa** [reply]
🤔 Honestly, I suppose I'm not 100% sure, so I'll take a step back for now, just to be safe. Good luck :) And thanks for the transparency.

**Reply by burggraf** [reply]
Absolutely.  And I have to say again, outstanding work.  You did stuff people didn't even think of trying!

**Reply by jmezzacappa** [reply]
Thank you! It has been one of my favorite projects, and I'm glad (and somewhat surprised, if I'm being honest) that the reception has been overwhelmingly positive.

**Comment by Darin755** [comment]
Sorry for the delayed reply. 

When you compile buildroot what files are in ```output/images```? There should be a iso and the compressed filesystem. Also you should look though the output if haven't already to see if there are obvious errors.

**Reply by burggraf** [reply]
Hi @Darin755!  Thanks for the help!
At this point we have a custom iso created, and things are working pretty well.  We even have networking enabled and can access the outside world.  So cool!

The big issue we're having now is mounting the 9p filesystem to /mnt.  Should I open up another issue for that?

**Reply by Darin755** [reply]
The standard iso mounts 9pfs to ```/home``` not ```/mnt```. You can reference the [fstab](https://github.com/Darin755/browser-buildroot/blob/027e856888d50bbc7008b50221be0323cf46a128/standard/board/browser_linux/rootfs_overlay/etc/fstab#L9) for an example.

**Reply by Darin755** [reply]
I wrote some documentation for this [here](https://github.com/Darin755/browser-linux/blob/main/docs/9pfs.md). I hope that it is helpful.
